dist/index.es.js

```diff
@@ -1573,10 +1558,33 @@ Grid.propTypes = {
+var styles$1 = { "uploader": "uploader_uploader__1RhpY", "titleLabel": "uploader_titleLabel__3dNsX", "title": "uploader_title__6XJq_", "error": "uploader_error__WDlNo", "fileUploader": "uploader_fileUploader__2jVN9", "imagePreview": "uploader_imagePreview__3VSdX", "clearImage": "uploader_clearImage__2rMOj", "info": "uploader_info__3NSEz", "file": "uploader_file__1ekDC", "submit": "uploader_submit__2mAIq", "heading": "uploader_heading__1Ox1s", "hide": "uploader_hide__1T76k", "errors": "uploader_errors__306yx", "errorMessage": "uploader_errorMessage__KwtDp" };
 styleInject(css$2);
 
+var validate = function validate(file, restrictions) {
+	var errors = [];
+	return new Promise(function (resolve, reject) {
+		if (!restrictions || !Object.keys(restrictions).length) return resolve();
+		if (restrictions.size && file.size > restrictions.size) errors.push("Uploaded image's size exceeds the accepted limit of " + Math.ceil(restrictions.size / 1000000) + " MB.");
+		if (restrictions.dimensions) {
+			var reader = new FileReader();
+			reader.onload = function (e) {
+				var img = new Image();
+				img.src = e.target.result;
+				img.onload = function () {
+					if (restrictions.dimensions.width !== 0 && this.width !== restrictions.dimensions.width) errors.push("Uploaded image's width does not match required value of " + restrictions.dimensions.width + " pixels.");
+					if (restrictions.dimensions.height !== 0 && this.height !== restrictions.dimensions.height) errors.push("Uploaded image's height does not match required value of " + restrictions.dimensions.height + " pixels.");
+					if (errors.length) reject(errors);else resolve();
+				};
+			};
+			reader.readAsDataURL(file);
+		} else {
```

Comment:
This will give unexpected behaviour if there is only size validation and no dimensions validation.